### PR TITLE
Remove the WithSDKOptions from the Jaeger exporter

### PR DESCRIPTION
**Pull request by MrAlias**
Removed the Jaeger exporter `WithSDKOptions` `Option`. This option was used to set SDK options for the exporter creation convenience functions. These functions are provided as a way to easily setup or install the exporter with what are deemed reasonable SDK settings for common use cases. If the SDK needs to be configured differently, the `NewRawExporter` function and direct setup of the SDK with the desired settings should be used.

Part of #1803

### CODE REVIEW

**Review by paivagustavo — APPROVED** [review]

File: exporters/trace/jaeger/jaeger_test.go [hunk 1]
```diff
@@ -823,49 +836,3 @@ func TestProcess(t *testing.T) {
...
-
-	testCollector := &testCollectorEndpoint{}
-	tp, err := NewExportPipeline(
-		withTestCollectorEndpointInjected(testCollector),
-		WithSDKOptions(
-			sdktrace.WithResource(resource.NewWithAttributes(
-				semconv.ServiceNameKey.String(serviceName),
-				attribute.String(tagKey, tagVal),
-			)),
-			sdktrace.WithSpanLimits(sdktrace.SpanLimits{
```
Comment (MrAlias) [inline exporters/trace/jaeger/jaeger_test.go:843]: This test was removed because the only thing this tests that `TestExporterExportSpan` doesn't is that it validates the SDK event limits are correctly handled by the SDK. This is not related to the exporter and is only an configuration option of the SDK so it was removed here.

File: example/jaeger/main.go [hunk 2]
```diff
@@ -21,26 +21,15 @@ import (
...
 )
 
 func main() {
 	// Create and install Jaeger export pipeline as the global.
 	tp, err := jaeger.InstallNewPipeline(
 		jaeger.WithCollectorEndpoint("http://localhost:14268/api/traces"),
-		jaeger.WithSDKOptions(
-			sdktrace.WithSampler(sdktrace.AlwaysSample()),
-			sdktrace.WithResource(resource.NewWithAttributes(
-				semconv.ServiceNameKey.String("trace-demo"),
```
Comment (Aneurysm9) [inline example/jaeger/main.go:39]: If a service name isn't set here, what service name is used?  The one from the default resource?  Should we use `NewRawExporter` here instead, or maybe have an option for adding a service name through this function?

**Review by Aneurysm9 — APPROVED** [review]

File: example/jaeger/main.go (hunk 2 above)
Comment (MrAlias) [reply example/jaeger/main.go:39]: Yeah, it will be the service name from the default resource. I'm okay with switching this to use the `NewRawExporter` and show how to set the default service name that way.

If we want to updated this to accept a service name we should probably do it for all the exporters. That work seems like it should be a separate PR/Issue if we want to go that direction.

Comment (MrAlias) [reply example/jaeger/main.go:39]: Updated. @Aneurysm9 PTAL

Comment (Aneurysm9) [reply example/jaeger/main.go:39]: LGTM
